src/types/redis_zset.cc

```diff
@@ -71,19 +72,35 @@ rocksdb::Status ZSet::Add(const Slice &user_key, uint8_t flags, std::vector<Memb
     }
     added_member_keys.insert(member_key);
 
+    bool lt = (flags & kZSetLT) != 0;
+    bool gt = (flags & kZSetGT) != 0;
+
     if (metadata.size > 0) {
       std::string old_score_bytes;
       s = db_->Get(rocksdb::ReadOptions(), member_key, &old_score_bytes);
       if (!s.ok() && !s.IsNotFound()) return s;
       if (s.ok()) {
+        if (!s.IsNotFound() && (flags & kZSetNX)) {
+          continue;
```

Comment:
Can just return instead of continuing here?

Reply:
It should not return here.
eg. 
del z1
zadd z1 1 a
zadd z1 nx 2 b 2 a  //  b should be added since it is new.

If it returns here, b has no chance to add.

Reply:
Yes, you're right, my bad for misunderstanding the break logic.